[awhb] iP

awhb opened the pull request:
# ChadBod Pro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

*ChadBod Pro* frees your mind of having to remember things you need to do. It's:

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is:

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

awhb committed:
Week 2 - Duke Level-0

awhb committed:
Week 2 - Duke Level-1

awhb committed:
Week 2 - Duke Level-2

awhb committed:
Week 2 - Duke Level-3

awhb committed:
Week 2 - Duke Level-4

awhb committed:
Week 2 - Duke A-TextUiTesting

awhb committed:
Week 2 - Duke Level-5

awhb committed:
Week 2 - Duke Level-6

awhb committed:
Week 2 - A-Enums

awhb committed:
Week 3 - Duke Level-7

awhb committed:
Week 3 - Duke Level-8

awhb committed:
Duke A-MoreOOP

awhb committed:
Week 3 - Duke A-MoreOOP

awhb committed:
Merge branch 'branch-Level-7'

awhb committed:
Merge branch 'branch-Level-8'

awhb committed:
Merge branch 'add-gradle-support'

awhb committed:
Week 3 - Duke A-Gradle

awhb committed:
Week 3 - A-JUnit

awhb committed:
Week 3 - Duke A-Jar

awhb committed:
Week 3 - Duke A-JavaDoc

awhb committed:
Week 3 - Duke A-CodingStandard

awhb committed:
Week 3 - Duke Level-9

awhb committed:
Merge branch 'branch-A-JavaDoc'

awhb committed:
Merge branch 'branch-A-CodingStandard'

awhb committed:
Merge branch 'branch-Level-9'

awhb committed:
Week 4 - A-JavaDoc

awhb committed:
Week 4 - Duke A-CheckStyle

awhb committed:
Merge branch 'branch-A-CheckStyle'

yanghengtang commented on src/main/java/chadbod/ChadBod.java:
```diff
@@ -0,0 +1,176 @@
+                ParsedCommand parsedCommand = Parser.parseCommand(input);
+                String details = parsedCommand.getDetails();
+                switch (parsedCommand.getCommand()) {
+                case BYE:
```
Well done for indent Switch Cases correctly!

yanghengtang commented on src/main/java/chadbod/ChadBod.java:
```diff
@@ -0,0 +1,176 @@
+        }
+    }
+
+    private int getTaskNumber(String details) throws NumberFormatException, TaskIndexOutOfBoundsException {
```
Is there a better name for `details` to communicate what the input parameter is?

papataco14 replied:
Yes I agree, a more descriptive name for details would be good! Perhaps TaskDescription?

yanghengtang commented on src/main/java/chadbod/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Using `byDate` as the attribute name would provide a better clarity.

yanghengtang commented on src/main/java/chadbod/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+     * @param i The index of the task to retrieve.
+     * @return The task at the specified index.
+     */
+    public Task getTask(int i) {
```
Using `i` as the parameter name might not communicate to users what input they should be placing into it. An alternative would be to use `index`

yanghengtang commented on src/main/java/chadbod/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+     *
+     * @param t The task to be added.
+     */
+    public void addTask(Task t) {
```
Likewise for the choice of parameter name here, `task` might better convey to the user that they are supposed to input.

yanghengtang reviewed (commented):
Well done on your progress on the chatbot!

awhb committed:
checked style and added comprehensiveness to test methods

awhb committed:
Week 4 - Duke Level-10

awhb committed:
Merge branch 'branch-Level-10'

awhb committed:
Week 4 - Removed text-ui-test

awhb committed:
Week 4 - Duke UI Improved

papataco14 commented on src/main/java/chadbod/ChadBod.java:
```diff
@@ -0,0 +1,178 @@
+ * The ChadBod class represents a task management application.
+ */
+public class ChadBod {
+    private static final String FILE_PATH = "./data/tasks.txt";
+    private static final int TASKLIST_DISPLAY_OFFSET = 1;
```
Good job on following the coding standards for final attributes!

papataco14 commented on src/main/java/chadbod/Command.java:
```diff
@@ -0,0 +1,36 @@
+    DELETE("delete"),
+    FIND("find");
+
+    private final String value;
```
Is there a more descriptive name for value?

papataco14 commented on src/main/java/chadbod/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public String toString() {
+        return String.format("[D]%s (by: %s)", super.toString(),
+                    this.by.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:mm:ss")));
+    }
```
I like your use of line breaks to not exceed line limit!

papataco14 reviewed (commented):
LGTM, just some nits to pick

awhb committed:
Generate a new JAR file

The current JAR file is based on the old version of the code, without
reflecting changes made to the code that include incorporating
JavaFX GUI.

The current JAR file is thus no longer suitable as the latest release.

Let's generate a new JAR file that reflects the latest changes,
which include incorporating JavaFX GUI.

This new JAR file will henceforth be used as the latest release.

awhb committed:
Include Java assertions in program code

Currently, there are no Java assertions in the program code.

It is hard to keep track of important assumptions in various parts of
the program as a result, so future regressions may be more easily
resolved.

Let's add Java assertions to the program code.

This will make it easier to track important assumptions in various
parts of the program and thus resolve future regressions.

awhb committed:
Week 5 - Duke A-CodeQuality

awhb committed:
Merge pull request #1 from awhb/branch-A-Assertions

Include Java assertions in program code

awhb committed:
Merge branch 'master' into branch-A-CodeQuality

awhb committed:
Merge pull request #2 from awhb/branch-A-CodeQuality

Improve code quality of project

awhb committed:
Week 5 - Duke A-CI

awhb committed:
Week 5 - Duke A-CI

awhb committed:
Week 5 - Duke BCD-Extension: C-FriendlierSyntax

awhb committed:
Merge branch 'branch-BCD-Extension'

awhb committed:
Reset data file

awhb committed:
Week 6 - Duke A-CodeQuality fixes

awhb committed:
Week 6 - Duke A-CodeQuality fixes

awhb committed:
Week 6 - Duke A-UserGuide

awhb committed:
Week 6 - Duke A-Release

awhb committed:
Fixes to pass CI tests

awhb committed:
Fixes to pass CI tests

awhb committed:
Week 6 - Update README files
